tests/rancher/rancher_ui.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+# Maintainer: George Gkioulis <ggkioulis@suse.com>
+
+use base 'x11test';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use x11utils 'ensure_unlocked_desktop';
+
+sub run {
+    my ($self) = shift;
+
+    # switch to desktop
+    if (!check_var('DESKTOP', 'textmode')) {
+        select_console('x11', await_console => 0);
+    }
+
+    ensure_unlocked_desktop();
+
+    # start firefox
+    $self->start_firefox_with_profile();
+    $self->firefox_open_url('https://localhost');
```

Comment:
I suggest to simplify the test by starting firefox with the argument `https://localhost` directly. Maybe you can already work around the "security risk warning" before even starting firefox to not have unnecessary, rather unrelated needle checks.

Also I don't get why you use an xterm. Why not use the desktop runner as in other test modules? Then you also don't need to cleanup any xterm instances.

Reply:
The firefox methods are actually required to deal with stuff like "I HAVE A NEW FEATURE" and "Default browser pls?".

Reply:
> Also I don't get why you use an xterm. Why not use the desktop runner as in other test modules? Then you also don't need to cleanup any xterm instances.

I don't have any preference in using xterm, it is just used by the firefox function that I call. I will probably try some alternative to minimize cleanup as well